Dont send small contracted invoices to finance

proteusvacuum opened the pull request:
finance@dimagi doesn't want to receive small invoices for contracted
clients any more.
http://manage.dimagi.com/default.asp?169520#972816

@nickpell 
CC @biyeun

proteusvacuum committed:
Dont send small contracted invoices to finance

finance@dimagi doesn't want to receive small invoices for contracted
clients any more.
http://manage.dimagi.com/default.asp?169520#972816

nickpell commented on corehq/apps/accounting/invoicing.py:
```diff
@@ -215,6 +215,12 @@ def create_invoice_for_subscription(self, subscription):
 
         return invoice
 
+    def should_send_email(self, invoice, subscription):
```
Should this function also take ```do_not_invoice``` into account?

proteusvacuum replied:
yes. will add that in a bit.

nickpell commented:
test failure looks related

proteusvacuum committed:
Move should_send_email to BillingRecord

This check was happening in multiple places, moved it all into
`BillingRecord`. I think that makes more sense.

dimagimon commented on corehq/apps/accounting/tests/test_invoice_factory.py:
```diff
@@ -1,7 +1,16 @@
 import datetime
+from decimal import Decimal
```
F401 'Decimal' imported but unused

dimagimon commented on corehq/apps/accounting/tests/test_invoice_factory.py:
```diff
@@ -1,7 +1,16 @@
 from corehq.apps.accounting import generator
 from corehq.apps.accounting.invoicing import DomainInvoiceFactory
-from corehq.apps.accounting.models import DefaultProductPlan, BillingAccount, Subscription, SubscriptionAdjustment
+from corehq.apps.accounting.models import (
```
F401 'Invoice' imported but unused

dimagimon replied:
F401 'SMALL_INVOICE_THRESHOLD' imported but unused

dimagimon replied:
F401 'SubscriptionType' imported but unused

proteusvacuum commented:
I think this is better.

proteusvacuum committed:
Update should_send_email to work with Wires

nickpell commented on corehq/apps/accounting/models.py:
```diff
@@ -1822,6 +1833,15 @@ def text_template(self):
+    @property
+    def should_send_email(self):
+        subscription = self.invoice.subscription
+        autogenerate = (subscription.auto_generate_credits and not self.invoice.balance)
```
(nonblocking) this variable name could be clearer (it took me some time to figure out what was going on here)
